app/src/main/java/com/github/se/gomeet/ui/mainscreens/events/EventInfo.kt

```diff
@@ -256,7 +256,8 @@ fun EventButtons(
               } else {
                 if (isJoined.value) {
                   Text("Leave Event")
-                } else {
+                } else if (currentEvent.value?.public == true ||
```

Comment:
This condition should be elsewhere or shouldn't be in this class at all: for one thing, it shouldn't be possible to get to this part of the code if the user can't see this event widget. For another, this condition only masks the text, it doesn't disable the TextButton.